src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -0,0 +1,646 @@
+
+    # Add qualitative similarity columns (fuel_type_code_pudl)
+    check_connections.loc[
+        (check_connections.fuel_type_code_pudl_eia.notna())
+        & (check_connections.fuel_type_code_pudl_ferc1.notna()),
+        "fuel_type_code_pudl_diff",
+    ] = check_connections.fuel_type_code_pudl_eia == (
+        check_connections.fuel_type_code_pudl_ferc1
+    )
+
+    # Add quantitative similarity columns (installation year)
+    check_connections.loc[
+        :, "installation_year_ferc1"
+    ] = check_connections.installation_year_ferc1.astype("Int64")
+    check_connections.loc[
+        (check_connections.installation_year_eia.notna())
+        & (check_connections.installation_year_ferc1.notna()),
+        "installation_year_diff",
+    ] = (
+        check_connections.installation_year_eia
+        - check_connections.installation_year_ferc1
+    )
+
+    # Move record_id_ferc1
+    record_id_ferc1 = check_connections.pop("record_id_ferc1")
+    check_connections.insert(9, "record_id_ferc1", record_id_ferc1)
+
+    # Add utility name eia
+    utils = (
+        pudl_out.utils_eia860()
+        .assign(report_year=lambda x: x.report_date.dt.year)[
+            ["utility_id_eia", "utility_name_eia", "report_year"]
+        ]
+        .copy()
+    )
+
+    check_connections = pd.merge(
+        check_connections.dropna(
+            subset=["record_id_ferc1"]
+        ),  # dropna for now becuase somehow there is a full NA row in ferc-eia....figure out later
+        utils,
+        on=["utility_id_eia", "report_year"],
+        how="left",
+        validate="m:1",
+    ).rename(columns={"utility_name_eia": "utility_name"})
+
+    check_connections.insert(19, "utility_name_eia", check_connections.utility_name)
+    check_connections = check_connections.drop(columns=["utility_name"])
+
+    return check_connections
+
+
+def _prep_ppl(ppl, pudl_out):
+    """Prep PPL for use in override output sheet pre-utility subgroups."""
+    logger.debug("Prepping Plant Parts Table")
+
+    # Add utilty name eia and only take relevant columns
+    ppl_out = (
+        ppl.reset_index()
+        .merge(
+            pudl_out.utils_eia860()[
+                ["utility_id_eia", "utility_name_eia", "report_date"]
+            ].copy(),
+            on=["utility_id_eia", "report_date"],
+            how="left",
+            validate="m:1",
+        )[RELEVANT_COLS_PPL]
+        .copy()
+    )
+
+    return ppl_out
+
+
+def _prep_deprish(deprish, pudl_out):
+    """Prep depreciation data for use in override output sheet pre-utility subgroups."""
+    logger.debug("Prepping Deprish Data")
+
+    # Get utility_id_eia from EIA
+    util_df = pudl_out.utils_eia860()
+    id_dict = dict(zip(util_df["utility_id_pudl"], util_df["utility_id_eia"]))
+
+    deprish_out = deprish.assign(
+        report_year=lambda x: x.report_date.dt.year.astype("Int64"),
+        utility_id_eia=lambda x: x.utility_id_pudl.map(id_dict).astype("Int64"),
```

Comment:
I'm pretty sure that I did it this way because there was some weirdness with NA values that merge didn't like.

Regarding m:1 for EIA and pudl ids, that is a valid concern...I think I thought that only applied to plant ids not utility ids. I could definitely change the system to rely on pudl ids instead, but I was nervous that those would change as we added more and then we'd have to re-map them. Maybe we could still use EIA utility ids, but then map those to pudl ids behind the scenes? I'm not sure...

Reply:
The merge version is also definitely faster, so I can use that in whatever fix we choose

Reply:
I would personally think about relying on the `utility_id_pudl` and just using a merge. As long as you don't have to hard code id's for specific utilities - because then yes you run into the issue of the pudl id changing over time. The times when I've had to hard code ID's I have hard coded the EIA ID but then just grab the associated pudl ID and use that for inter-dataset merges.

Reply:
👀  this map dict thing still feels potentially slow and less clean than just using a merge.